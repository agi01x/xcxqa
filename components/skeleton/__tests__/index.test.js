import React from 'react';
import { mount } from 'enzyme';
import Skeleton from '..';

describe('Skeleton', () => {
  const genSkeleton = props => mount(
    <Skeleton loading {...props}>
      Bamboo
    </Skeleton>
  );

  describe('avatar', () => {
    it('size', () => {
      const wrapperSmall = genSkeleton({ avatar: { size: 'small' } });
      expect(wrapperSmall.render()).toMatchSnapshot();
      const wrapperDefault = genSkeleton({ avatar: { size: 'default' } });
      expect(wrapperDefault.render()).toMatchSnapshot();
      const wrapperLarge = genSkeleton({ avatar: { size: 'large' } });
      expect(wrapperLarge.render()).toMatchSnapshot();
    });

    it('shape', () => {
      const wrapperCircle = genSkeleton({ avatar: { shape: 'circle' } });
      expect(wrapperCircle.render()).toMatchSnapshot();
      const wrapperSquare = genSkeleton({ avatar: { shape: 'square' } });
      expect(wrapperSquare.render()).toMatchSnapshot();
    });
  });

  describe('title', () => {
    it('width', () => {
      const wrapper = genSkeleton({ title: { width: '93%' } });
      expect(wrapper.render()).toMatchSnapshot();
    });
  });

  describe('paragraph', () => {
    it('rows', () => {
      const wrapper = genSkeleton({ paragraph: { rows: 5 } });
      expect(wrapper.render()).toMatchSnapshot();
    });

    it('width', () => {
      const wrapperPure = genSkeleton({ paragraph: { width: '93%' } });
      expect(wrapperPure.render()).toMatchSnapshot();
      const wrapperList = genSkeleton({ paragraph: { width: ['28%', '93%'] } });
      expect(wrapperList.render()).toMatchSnapshot();
    });
  });
});
