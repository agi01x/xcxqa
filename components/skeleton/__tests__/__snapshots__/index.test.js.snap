// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton avatar shape 1`] = `
<div
  class="ant-skeleton ant-skeleton-with-avatar"
>
  <div
    class="ant-skeleton-header"
  >
    <span
      class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
    />
  </div>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 50%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
    </ul>
  </div>
</div>
`;

exports[`Skeleton avatar shape 2`] = `
<div
  class="ant-skeleton ant-skeleton-with-avatar"
>
  <div
    class="ant-skeleton-header"
  >
    <span
      class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-square"
    />
  </div>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 50%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
    </ul>
  </div>
</div>
`;

exports[`Skeleton avatar size 1`] = `
<div
  class="ant-skeleton ant-skeleton-with-avatar"
>
  <div
    class="ant-skeleton-header"
  >
    <span
      class="ant-skeleton-avatar ant-skeleton-avatar-sm ant-skeleton-avatar-circle"
    />
  </div>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 50%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
    </ul>
  </div>
</div>
`;

exports[`Skeleton avatar size 2`] = `
<div
  class="ant-skeleton ant-skeleton-with-avatar"
>
  <div
    class="ant-skeleton-header"
  >
    <span
      class="ant-skeleton-avatar ant-skeleton-avatar-circle"
    />
  </div>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 50%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
    </ul>
  </div>
</div>
`;

exports[`Skeleton avatar size 3`] = `
<div
  class="ant-skeleton ant-skeleton-with-avatar"
>
  <div
    class="ant-skeleton-header"
  >
    <span
      class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
    />
  </div>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 50%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
    </ul>
  </div>
</div>
`;

exports[`Skeleton paragraph rows 1`] = `
<div
  class="ant-skeleton"
>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 38%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
      <li />
      <li />
      <li
        style="width: 61%;"
      />
    </ul>
  </div>
</div>
`;

exports[`Skeleton paragraph width 1`] = `
<div
  class="ant-skeleton"
>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 38%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
      <li
        style="width: 93%;"
      />
    </ul>
  </div>
</div>
`;

exports[`Skeleton paragraph width 2`] = `
<div
  class="ant-skeleton"
>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 38%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li
        style="width: 28%;"
      />
      <li
        style="width: 93%;"
      />
      <li />
    </ul>
  </div>
</div>
`;

exports[`Skeleton title width 1`] = `
<div
  class="ant-skeleton"
>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 93%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
      <li
        style="width: 61%;"
      />
    </ul>
  </div>
</div>
`;
